src/foam/comics/DAOControllerView.js

```diff
@@ -21,52 +21,86 @@ foam.CLASS({
   name: 'DAOControllerView',
   extends: 'foam.u2.View',
 
-  imports: [
-    'stack'
+  requires: [
+    'foam.comics.DAOController'
   ],
 
-  exports: [
-    'editRecord'
+  imports: [
+    'stack',
+    'data? as importedData'
   ],
 
-  requires: [
-    'foam.comics.DAOController'
+  exports: [
+    'data.selection as selection',
+    'data.data as dao'
   ],
 
   properties: [
-    'data',
     {
-      name: 'controller',
-      factory: function() {
-        var controller = this.DAOController.create();
-        this.onDetach(controller.data$.follow(this.data$));
-        return controller;
-      }
+      class: 'FObjectProperty',
+      of: 'foam.comics.DAOController',
+      name: 'data',
+      expression: function(importedData) { return importedData; },
+    },
+    {
+      name: 'cls',
+      expression: function(data) { return data.cls_; }
     }
   ],
 
+  reactions: [
+    [ 'data', 'action,create', 'onCreate' ],
+    [ 'data', 'edit', 'onEdit' ],
+    [ 'data', 'action,findRelatedObject', 'onFindRelated' ],
+    [ 'data', 'finished', 'onFinished']
+  ],
+
   methods: [
-    function editRecord(obj) {
-      this.stack.push({
-        class: 'foam.comics.DAOUpdateControllerView',
-        dao: this.data,
-        data: obj.id
-      });
-    },
     function initE() {
-      this.startContext({ data: this.controller }).
+      this.
         start('table').
-          start('tr').
-            start('td').
-              start(this.DAOController.PREDICATE, {dao$: this.data$}).end().
-            end().
-            start('td').style({ 'vertical-align': 'top', 'width': '100%' }).add(this.DAOController.FILTERED_DAO).end().
-          end().
         start('tr').
-          show(this.mode$.map(function(m) { return m == foam.u2.DisplayMode.RW; })).
-          start('td').end().start('td').add(this.DAOController.CREATE).end().
+          start('td').add(this.cls.PREDICATE).end().
+          start('td').style({ 'vertical-align': 'top', 'width': '100%' }).
+            add(this.cls.FILTERED_DAO).end().
```

Comment:
Line the end() up with the start() if they aren't on the same line, otherwise it looks like a bug at first.